### PR TITLE
Make it possible to skip following stages when plan result is no changes

### LINKED ISSUE #3749: Make it possible to skip following stages when the plan result is 'No changes'
**What would you like to be added**:
It is very useful to skip WAIT_APPROVAL and the following stages when the plan result is No changes.
Add an option to StageTerraformPlan for it.

**Why is this needed**:

### PR DESCRIPTION
**Pull request by golemiso**
This is a proposed solution for skipping stages after plan.
I am not an expert on this product so this can break existing workflow. Should be integration-tested.

**Which issue(s) this PR fixes**:

- Fixes #3749

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by nghialv** [comment]
@golemiso Hi, Thank you for your PR. 🎊

File: pkg/app/piped/executor/terraform/deploy.go [hunk 1]
```diff
@@ -152,31 +158,31 @@ func (e *deployExecutor) ensurePlan(ctx context.Context) model.StageStatus {
...
 	if err != nil {
 		e.LogPersister.Errorf("Failed to plan (%v)", err)
-		return model.StageStatus_STAGE_FAILURE
+		return model.StageStatus_STAGE_FAILURE, provider.PlanResult{}
 	}
 
 	if planResult.NoChanges() {
 		e.LogPersister.Success("No changes to apply")
-		return model.StageStatus_STAGE_SUCCESS
+		return model.StageStatus_STAGE_SUCCESS, planResult
```
Comment (knanao) [inline pkg/app/piped/executor/terraform/deploy.go:181]: How about this without changing the return value?
```suggestion
if e.StageConfig.TerraformPlanStageOptions.ExitOnNoChanges {
        return model.StageStatus_STAGE_EXITING
}
```

**Review by knanao — COMMENTED** [review]
@golemiso 
I'm so sorry for the late reply and thank you a lot for your contributions.
I left some nits comments so I'm glad for you to check it.

**Review by knanao — APPROVED** [review]
Nice work.
Please let me take over the rest.

File: pkg/app/piped/executor/terraform/deploy.go [hunk 2]
```diff
@@ -81,7 +81,7 @@ func (e *deployExecutor) Execute(sig executor.StopSignal) model.StageStatus {
 		status = e.ensureApply(ctx)
 
 	default:
-		e.LogPersister.Errorf("Unsupported stage %s for cloudrun application", e.Stage.Name)
+		e.LogPersister.Errorf("Unsupported stage %s for terraform application", e.Stage.Name)
```
Comment (nghialv) [inline pkg/app/piped/executor/terraform/deploy.go:84]: Nice catch!

**Comment by nghialv** [comment]
@golemis Thank you for your great PR. Can't wait to have this in the next release.

**Pull request merged**